activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -920,6 +921,10 @@ def prepare_statement(sql, binds)
         # connected server's characteristics.
         def connect
           @raw_connection = self.class.new_client(@connection_parameters)
+
+          unless ActiveRecord.db_warnings_action == :ignore
+            @raw_connection.set_notice_processor { |message| @notice_processor_sql_warnings << message }
+          end
```

Comment:
```suggestion
          @raw_connection.set_notice_processor { |message| @notice_processor_sql_warnings << message }
```

Same here.

Reply:
Ah, right. This means we'll always overwrite the existing notice processor if there is one, but I'm not sure how widely this feature is used (and arguably, the SQL warnings API should serve as a replacement). Any concerns here?

Reply:
> I think we'll always be setting it before any existing code

I think you're completely right -- though maybe then we ought to worry about them clobbering _our_ block if they do end up setting a notice processor | receiver. Anyways, this seems unlikely, so I'll leave it for now :)